code/modules/shuttle/navigation_computer.dm

```diff
@@ -8,12 +8,15 @@
 	var/shuttlePortId = ""
 	var/shuttlePortName = "custom location"
 	var/list/jumpto_ports = list() //hashset of ports to jump to and ignore for collision purposes
+	var/list/beacon_codes = list() //hashset of beacon to jump to
+	var/list/beacon_access_codes = list() //hashset of access codes to beacons
 	var/obj/docking_port/stationary/my_port //the custom docking port placed by this console
 	var/obj/docking_port/mobile/shuttle_port //the mobile docking port of the connected shuttle
+	var/list/z_locked = list(0) //z forbided for custom docking
```

Comment:
Do not use constant number values for referencing z levels. Strictly speaking we dont enforce maps on specific z levels.

Reply:
About this 0, this is block for all possibilities to jump cam to the null, i think this can be possible on beacot qdel. Its fill by sorting all map traits what we want.

Reply:
Now use trait blacklist